Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift

```diff
@@ -0,0 +1,186 @@
+import UIKit
+import MapboxCommon
+import MapboxCoreMaps
+
+extension GestureManager: GestureHandlerDelegate {
+    // MapView has been tapped a certain number of times
+    internal func tapped(numberOfTaps: Int, numberOfTouches: Int) {
+
+        guard let mapView = cameraManager.mapView else {
+            return
+        }
+
+        // Single tapping twice with one finger will cause the map to zoom in
+        if numberOfTaps == 2 && numberOfTouches == 1 {
+            cameraManager.setCamera(to: CameraOptions(zoom: mapView.zoom + 1.0),
+                                    animated: true,
+                                    duration: 0.3,
+                                    completion: nil)
+        }
+
+        // Double tapping twice with two fingers will cause the map to zoom out
+        if numberOfTaps == 2 && numberOfTouches == 2 {
+            cameraManager.setCamera(to: CameraOptions(zoom: mapView.zoom - 1.0),
+                                    animated: true,
+                                    duration: 0.3,
+                                    completion: nil)
+        }
+    }
+
+    internal func panBegan(at point: CGPoint) {
+        try! cameraManager.mapView?.__map.dragStart(forPoint: point.screenCoordinate)
+    }
+
+    // MapView has been panned
+    internal func panned(from startPoint: CGPoint, to endPoint: CGPoint) {
+
+        if let cameraOptions = try! cameraManager.mapView?.__map.getDragCameraOptionsFor(fromPoint: startPoint.screenCoordinate, toPoint: endPoint.screenCoordinate) {
+
+            cameraManager.setCamera(to: cameraOptions,
+                                    animated: false,
+                                    duration: 0,
+                                    completion: nil)
+
+        }
+    }
+
+    // Pan has ended on the MapView with a residual `offset`
+    func panEnded(at endPoint: CGPoint, shouldDriftTo driftEndPoint: CGPoint) {
+
+        if endPoint != driftEndPoint,
+           let driftCameraOptions = try! cameraManager.mapView?.__map.getDragCameraOptionsFor(fromPoint: endPoint.screenCoordinate, toPoint: driftEndPoint.screenCoordinate) {
+
+            cameraManager.setCamera(to: driftCameraOptions,
+                                    animated: true,
+                                    duration: Double(UIScrollView.DecelerationRate.normal.rawValue),
```

Comment:
We should probably use `cameraManager.mapCameraOptions.decelerationRate` instead of scroll view in case if the app developer provided a specific deceleration rate

Reply:
yep good point!